### ISSUE TITLE
hasStyle does not work for CSS variables

### ISSUE THREAD

**Issue by DingoEatingFuzz** [mainmatter/qunit-dom#1272]
## The bug

Consider this element:

```html
<div style="--my-var:123" />
```

The following assertion does not work:

```js
assert.dom('div').hasStyle({ '--my-var': 123 });
```

Since `hasStyle` is implemented with `getComputedStyle` and `getComputedStyle` returns styles after computations are resolved, variable styles are not present on the `CSSDeclaration` object.

In order to get the variable styles from an element you must use `element.style.getPropertyValue('--my-var')`.

## Motivation to have this behavior

An emerging pattern now that CSS variables have broad adoption is to dynamically set variables in JS/HTML and then read the variables in static CSS.

For instance:

```css
div {
  --my-var: 100; /* default value */
  height: var(--my-var); /* overridable from JS without hardcoding height */
  max-height: var(--my-var); /* implementation detail not the concern of JS */
}
```

## Proposed solution

I think the API for `hasStyle` can remain unchanged. The keys for the `expected` object can be divided into variables (start with `--`) and declarations. Then the two sets can be iterated over using `style.getPropertyValue` and `getComputedStyle` respectively.

**Comment by monovertex** [comment]
I came upon this feature request while looking for the same functionality. If this is something wanted in the proposed form I can take a stab at implementing this and opening a PR.

**Comment by vladucu** [comment]
We would love to see support added for this.
An attempt to implement this already exist in #1972. Anything I can do to get this merged?

**Comment by BobrImperator** [comment]
3.1.0 has been released https://www.npmjs.com/package/qunit-dom/v/3.1.0

See if it works for you

**Comment by BobrImperator** [comment]
Closing this since we've released the 3.1.0 version, if the issue is still present, feel free to reopen.
